File: desktopclient/src/main/java/com/goexp/galgame/gui/db/mysql/query/CVQuery.java
package com.goexp.galgame.gui.db.mysql.query;

import com.goexp.common.db.mysql.DBQueryTemplate;
import com.goexp.galgame.common.model.CV;
import com.goexp.galgame.gui.db.ICVQuery;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class CVQuery extends DBQueryTemplate<CV> implements ICVQuery {

    @Override
    public List<CV> cvList() {


        try {
            return runner.query("SELECT *  FROM `cv`", resultSet -> {
                var list = new ArrayList<CV>();
                while (resultSet.next()) {
                    var cv = new CV();
                    cv.star = resultSet.getInt("star");
                    cv.name = resultSet.getString("name");
                    cv.nameStr = resultSet.getString("names");
                    list.add(cv);
                }

                return list;
            });
        } catch (SQLException e) {
            e.printStackTrace();
        }

        return null;

    }
}

File: webapi/app/qurey/CVQuery.scala
package qurey

import com.goexp.db.mongo.{DBQuery, ObjectCreator}
import com.goexp.galgame.common.Config
import com.goexp.galgame.common.Config.DB_NAME
import com.goexp.galgame.common.model.CV
import com.typesafe.scalalogging.Logger
import org.bson.Document

import java.util
import scala.jdk.CollectionConverters._

object CVQuery {
  private val tlp = DBQuery[CV](Config.DB_STRING, DB_NAME, "cv", CVCreator).build

  def apply() = tlp


  private object CVCreator extends ObjectCreator[CV] {

    private val logger = Logger(CVCreator.getClass)


    override def create(doc: Document): CV = {
      logger.trace(s"<Doc> $doc")

      val cv = new CV
      cv.id = doc.getInteger("_id")
      cv.name = doc.getString("name")
      cv.star = doc.getInteger("star")

      cv.tag = Option(doc.get("tag", classOf[util.List[String]])).map { l => l.asScala.toList }.orNull
      cv.nameStr = doc.getString("names")


      cv.statistics = Option(doc.get("statistics", classOf[Document])).map(StatCreators.statisticsCreator.create).orNull

      logger.trace(s"<CV> $cv")

      cv
    }
  }
}